precompile/contract/interfaces.go

```diff
@@ -51,19 +51,24 @@ type AccessibleState interface {
 	GetSnowContext() *snow.Context
 }
 
-// BlockContext defines an interface that provides information to a stateful precompile about the
-// current block. The BlockContext may be provided during both precompile activation and execution.
-type BlockContext interface {
+// ConfigurationBlockContext defines the interface required to configure a precompile.
+type ConfigurationBlockContext interface {
 	Number() *big.Int
 	Timestamp() uint64
 }
 
+// BlockContext defines the interface to provide information about the block to a precompile
+// during the execution of the EVM.
+type BlockContext interface {
+	ConfigurationBlockContext
```

Comment:
I know we will add more functions to this interface. but still, is there any reason why we won't just keep expanding the old one?

Reply:
We use `*types.Block` as the `ConfigurationBlockContext` where only `Time()` and `Number()` are required. Separating this out into the interface required for configuring the precompile vs. requirements at execution time allow us to keep passing the block rather than constructing the block context there, which requires the `ChainContext` that is used to serve `GetBlockHashFn`. Specifically, constructing the `BlockContext` here https://github.com/ava-labs/subnet-evm/blob/master/core/genesis.go#L304 would require the `ChainContext`, so I opted to separate the block context required for configuration vs. execution because they serve slightly different purposes.